### PR TITLE
feat: add email registration check and StartFragment in AuthActivity

### LINKED ISSUE #1238: Add Email Check and Get Started screen
**Is it a ?**

- [x] Feature Request
    
### Feature Request

**Feature Description**

An email check to check if the email is registered in the Get Started screen was needed as per the Action Items list in meeting notes.

**Parent issue**
#971 
 
**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mishuvs**
Fixes #1238 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream development branch.

Changes: Email check in a new Get Started screen, same as in the EventBrite app. If the email is registered, transitions to LogIn screen otherwise SignUp screen.

Screenshots for the change:
videotogif_2018 07 22_17 46 43

### CODE REVIEW

File: app/src/test/java/org/fossasia/openevent/app/core/auth/login/LoginViewModelTest.java [hunk 1]
```diff
@@ -107,49 +107,12 @@ public void shouldShowLoginError() {
...
     public void shouldNotLoginAutomatically() {
         Mockito.when(authModel.isLoggedIn()).thenReturn(false);
 
-        loginViewModel.getLoginStatus().observeForever(login);
-
-        verify(login, Mockito.never()).onChanged(true);
-    }
-
-    @Test
-    public void shouldAttachEmailAutomatically() {
```
Comment (iamareebjamal) [inline app/src/test/java/org/fossasia/openevent/app/core/auth/login/LoginViewModelTest.java]: This should be tested?

Comment (mishuvs) [reply app/src/test/java/org/fossasia/openevent/app/core/auth/login/LoginViewModelTest.java]: I am adding test for StartViewModel. I will include this test in it. In login screen if the email is changed, it transitions back to the Start screen as in EventBrite so email list is not required in Login Screen now.

**Comment by iamareebjamal** [comment]
I've been seeing this for few days, looks OK to me but I can't approve until at least 2 of @Masquerade0097 @sridharjajoo @srv-twry have approved because I haven't tested the features

**Comment by mishuvs** [comment]
@Masquerade0097 @sridharjajoo @srv-twry Please review

**Comment by mishuvs** [comment]
@Masquerade0097 I found about 2 of the issues yesterday only. I myself was thinking that it might create navigation issues. I was thinking if we should remove the SignUp or Login links from each page or not. Because EventBrite also doesn't have them in respective links.
Apologies, I should have given some more time for testing.
Instead of handling complex logic for handling backstack and transition, it's better to keep the navigation flat like EventBrite. I'll send an update soon.

**Comment by mishuvs** [comment]
@Masquerade0097 Please test and review again.

**Comment by mishuvs** [comment]
@sridharjajoo Please review this one as well.

**Comment by mishuvs** [comment]
@Masquerade0097 How much longer will it take to review the updates? You already have my branch locally. I need it merged before the next meeting. You only need to pull it and check the newer commits as you have already reviewed the older ones.

**Comment by Masquerade0097** [comment]
@mishuvs Apologies for the delay in reviewing, I got occupied in some stuff and reviewed other PR's but forgot this one somehow.
Because you need to finalize it asap and the code looks good to me (except for a small bug which can be worked on later), I'm approving it for now :)

File: app/src/main/java/com/eventyay/organizer/core/auth/login/LoginFragment.java [hunk 2]
```diff
@@ -55,11 +55,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onStart() {
         super.onStart();
 
-        loginFragmentViewModel.getLoginStatus().observe(this, (loginStatus) -> handleIntent());
         loginFragmentViewModel.getProgress().observe(this, this::showProgress);
         loginFragmentViewModel.getError().observe(this, this::showError);
-        loginFragmentViewModel.getEmailList().observe(this, this::attachEmails);
-        loginFragmentViewModel.getLogin().observe(this, login -> {
+        loginFragmentViewModel.getActionLogIn().observe(this, isLoggedIn -> handleIntent());
+        loginFragmentViewModel.getLogin(sharedViewModel.getEmail().getValue()).observe(this, login -> {
```
[the remark this answers is not in the document]
Comment (mishuvs) [reply app/src/main/java/com/eventyay/organizer/core/auth/login/LoginFragment.java]: Yes

**Comment by mishuvs** [comment]
@Masquerade0097 `the code looks good to me`, but you have tested right? What's the bug?
@srv-twry Please review

**Pull request merged**